### PR TITLE
refactor(lib/data): move metadata-manual to lib/data

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- move metadata-manual (ie. changelog & source url data) to `lib/data`
- divide the `changelogUrl` and `sourceUrl` data into separate json files
- update `lib/data/readme`

## Context
Part Of: https://github.com/renovatebot/renovate/issues/29942

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or https://github.com/Rahul-renovate-testing/changelog-source-url
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/data/readme.md [hunk 1]
```diff
@@ -53,3 +55,44 @@ Example:
...
+
+## Tell Renovate where to find changelogs (`changelog-urls.json`)
+
+The `changelog-urls.json` has all the changelog information.
+
+Renovate nearly always finds, and displays, the changelog for a package update automatically.
+
+To find the changelog, Renovate needs the:
+
+- Name of the package
```
Comment (secustor) [inline lib/data/readme.md:67]: Reference again the field name `packageName`, that the URLs are grouped by datasource and an example to explain this.

File: lib/data/readme.md [hunk 2]
```diff
@@ -53,3 +55,44 @@ Example:
...
+
+The `source-urls.json` has the information on source URL of multiple packages.
+
+Renovate nearly always finds, and displays, the source for a package update automatically.
+Usually, the API for the package to be updated gives Renovate the correct info.
+If this does not happen, for whatever reason, Renovate can not link to the source of the package and might not be able to lookup changelogs.
+
+To find the source URL, Renovate needs the:
+
+- Name of the package
```
Comment (secustor) [inline lib/data/readme.md:89]: Same here.

**Review by viceice — COMMENTED** [review]
LGTM beside the mentioned docs changes

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
@secustor I'm leaving one unresolved conversation of yours so that you have a chance to review

**Review by secustor — APPROVED** [review]

**Pull request merged**